### PR TITLE
[th2-2743] added bookName to Event

### PR DESCRIPTION
**Pull request by andreydrobynin**
Related to https://github.com/th2-net/th2-conn/pull/95

### CODE REVIEW

File: src/main/java/com/exactpro/th2/common/event/Event.java [hunk 1]
```diff
@@ -110,6 +112,16 @@ public static Event from(Instant startTimestamp) {
                 .build();
     }
 
+    public Event bookName(String bookName) {
+        if (isNotBlank(bookName)) {
+            if (this.bookName != null) {
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/event/Event.java:117]: The default value of this.book is DEFAULT_BOOK_NAME than this check will be false always.
Let's use the DEFAULT_BOOK_NAME value in the method which converted to proto.

File: src/main/java/com/exactpro/th2/common/event/Event.java [hunk 2]
```diff
@@ -244,8 +256,11 @@ public Event messageID(MessageID attachedMessageID) {
      * @deprecated prefer to use full object instead of part of them, use the {@link #toListProto(EventID)} method
      */
     @Deprecated
-    public List<com.exactpro.th2.common.grpc.Event> toProtoEvents(@Nullable String parentID) throws JsonProcessingException {
-        return toListProto(toEventID(parentID));
+    public List<com.exactpro.th2.common.grpc.Event> toProtoEvents(
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/event/Event.java:259]: This is part of a public API and I'm not sure about increasing the major version. Let's create a separate override method to pass additional parametres

File: src/main/java/com/exactpro/th2/common/event/Event.java [hunk 3]
```diff
@@ -271,15 +289,15 @@ public Event messageID(MessageID attachedMessageID) {
                     .append(description);
         }
         var eventBuilder = com.exactpro.th2.common.grpc.Event.newBuilder()
-                .setId(toEventID(id))
+                .setId(toEventID(bookName, id))
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/event/Event.java:292]: Let's use the default if null method from the StringUtil class

File: src/main/java/com/exactpro/th2/common/event/EventUtils.java [hunk 4]
```diff
@@ -36,12 +36,13 @@ public static Message createMessageBean(String text) {
         return new MessageBuilder().text(text).build();
     }
 
-    @Contract("null -> null; !null -> !null")
-    public static @Nullable EventID toEventID(@Nullable String id) {
+    @Contract("_, null -> null; _, !null -> !null")
+    public static @Nullable EventID toEventID(String bookName, @Nullable String id) {
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/event/EventUtils.java:40]: This is part of a public API please add a separate override method.

File: src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java [hunk 5]
```diff
@@ -627,29 +627,32 @@ public InputStream readDictionary(DictionaryType dictionaryType) {
     }
 
     /**
-     * If root event does not exist, it creates root event with its name = box name and timestamp
+     * If root event does not exist, it creates root event with its book name = box book name and name = box name and timestamp
      * @return root event id
      */
     @Nullable
-    public String getRootEventId() {
+    public EventID getRootEventId() {
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java:634]: It's also a public API

File: src/main/kotlin/com/exactpro/th2/common/schema/message/MessageRouterUtils.kt [hunk 6]
```diff
@@ -27,25 +27,33 @@ import com.exactpro.th2.common.grpc.EventBatch
 import com.exactpro.th2.common.grpc.MessageGroupBatch
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/MessageRouterUtils.kt:27]: This file contains a public API

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

File: src/main/java/com/exactpro/th2/common/event/Event.java (hunk 1 above)
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/event/Event.java:116]: As we discussed I removed default book name

File: src/main/java/com/exactpro/th2/common/event/Event.java (hunk 2 above)
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/event/Event.java]: Discussed in chat

File: src/main/java/com/exactpro/th2/common/event/EventUtils.java (hunk 4 above)
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/event/EventUtils.java:40]: Discussed in chat

File: src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java (hunk 5 above)
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java:634]: Discussed in chat

File: src/main/kotlin/com/exactpro/th2/common/schema/message/MessageRouterUtils.kt (hunk 6 above)
Comment (andreydrobynin) [reply src/main/kotlin/com/exactpro/th2/common/schema/message/MessageRouterUtils.kt:28]: Discussed in chat

File: src/main/java/com/exactpro/th2/common/event/Event.java (hunk 3 above)
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/event/Event.java]: Discussed in chat

File: src/main/kotlin/com/exactpro/th2/common/schema/message/MessageRouterUtils.kt [hunk 7]
```diff
@@ -24,34 +24,32 @@ import com.exactpro.th2.common.event.Event.Status.PASSED
...
         PUBLISH.toString(),
         EVENT.toString()
     )
 }
 
 fun MessageRouter<EventBatch>.storeEvent(
-    parentBookName: String,
-    parentId: String,
+    parentId: EventID,
+    bookName: String,
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/MessageRouterUtils.kt:49]: It is legal when an event from book A refers to an event from book B as a parent?

Comment (andreydrobynin) [reply src/main/kotlin/com/exactpro/th2/common/schema/message/MessageRouterUtils.kt]: took bookName from parentId

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**